tests/unit/hdKey.test.ts

```diff
@@ -61,4 +61,106 @@ describe("Hierarchical Deterministic Key (hdkey)", () => {
       expect(isValidBIP44Path("m/44'/637'/0/0/0/")).toBe(false);
     });
   });
+
+  // testing against https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-1-for-ed25519
+  describe("Ed25519", () => {
+    const ed25519 = [
+      {
+        seed: Hex.fromHexInput("000102030405060708090a0b0c0d0e0f"),
+        vectors: [
+          {
+            chain: "m",
+            private: "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7",
+            public: "00a4b2856bfec510abab89753fac1ac0e1112364e7d250545963f135f2a33188ed",
+          },
+          {
+            chain: "m/0'",
+            private: "68e0fe46dfb67e368c75379acec591dad19df3cde26e63b93a8e704f1dade7a3",
+            public: "008c8a13df77a28f3445213a0f432fde644acaa215fc72dcdf300d5efaa85d350c",
+          },
+          {
+            chain: "m/0'/1'",
+            private: "b1d0bad404bf35da785a64ca1ac54b2617211d2777696fbffaf208f746ae84f2",
+            public: "001932a5270f335bed617d5b935c80aedb1a35bd9fc1e31acafd5372c30f5c1187",
+          },
+          {
+            chain: "m/0'/1'/2'",
+            private: "92a5b23c0b8a99e37d07df3fb9966917f5d06e02ddbd909c7e184371463e9fc9",
+            public: "00ae98736566d30ed0e9d2f4486a64bc95740d89c7db33f52121f8ea8f76ff0fc1",
+          },
+          {
+            chain: "m/0'/1'/2'/2'",
+            private: "30d1dc7e5fc04c31219ab25a27ae00b50f6fd66622f6e9c913253d6511d1e662",
+            public: "008abae2d66361c879b900d204ad2cc4984fa2aa344dd7ddc46007329ac76c429c",
+          },
+          {
+            chain: "m/0'/1'/2'/2'/1000000000'",
+            private: "8f94d394a8e8fd6b1bc2f3f49f5c47e385281d5c17e65324b0f62483e37e8793",
+            public: "003c24da049451555d51a7014a37337aa4e12d41e485abccfa46b47dfb2af54b7a",
+          },
+        ],
+      },
+    ];
+
+    ed25519.forEach(({ seed, vectors }) => {
+      vectors.forEach(({ chain, private: privateKey }) => {
+        it(`should generate correct key pair for ${chain}`, () => {
+          // eslint-disable-next-line @typescript-eslint/dot-notation
+          const key = Ed25519PrivateKey["fromDerivationPathInner"](chain, seed.toUint8Array());
```

Comment:
wondering why not dot?  Is that cause the private function?

Swear you're a magician at typescript :)

Reply:
yep! private static method